src/components/HighTable/HighTable.tsx

```diff
@@ -141,21 +141,32 @@ export function HighTableInner({
   const { numRows } = data
   const { enterCellsNavigation, setEnterCellsNavigation, onTableKeyDown: onNavigationTableKeyDown, onScrollKeyDown, cellPosition, focusFirstCell } = useCellsNavigation()
   const { containerRef } = usePortalContainer()
-  const { setAvailableWidth } = useColumnWidths()
+  const { setAvailableWidth, setScrollNeedInputs, needsHorizontalScrollDebounced } = useColumnWidths()
   const { isHiddenColumn } = useColumnVisibilityStates()
   const { orderBy, onOrderByChange } = useOrderBy()
   const { selectable, toggleAllRows, pendingSelectionGesture, onTableKeyDown: onSelectionTableKeyDown, allRowsSelected, isRowSelected, toggleRowNumber, toggleRangeToRowNumber } = useSelection()
   const allColumnsParameters = useColumnParameters()
   // local state
   const [rowsRange, setRowsRange] = useState<RowsRange>({ start: 0, end: 0 })
   const [lastCellPosition, setLastCellPosition] = useState(cellPosition)
+  const [availableScrollerWidth, setAvailableScrollerWidth] = useState<number | undefined>(undefined)
 
   const columnsParameters = useMemo(() => {
     return allColumnsParameters.filter((_, index) => {
       return !isHiddenColumn?.(index)
     })
   }, [allColumnsParameters, isHiddenColumn])
 
+  const visibleColumnIndexes = useMemo(() => {
```

Comment:
nit, but for consistency, 
```suggestion
  const columnIndexes = useMemo(() => {
```